### DISCUSSION TITLE
Some of my packages aren't being pinned

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [renovatebot/renovate#14966] [Request Help] [answered]
So for some reason some of my packages aren't getting pinned by the github bot can anyone explain why?

https://github.com/The-Code-Monkey/TechStack/blob/master/packages/styled-system/package.json 

this is one of them

**Comment by The-Code-Monkey** [comment]
renovate also keeps trying to rollback my dependencies for some reason not sure why.

**Comment by rarkins** [comment, marked as answer]
You have set `:autodetectPinVersions`. The way that works is Renovate pins if it's confident the package.json is for an application, but does not pin if it thinks it might be a library. If you are sure what you want then you should set `rangeStrategy=pin` in your config.
